src/Stormpath/ClientBuilder.php

```diff
@@ -265,7 +285,14 @@ public function build()
 
         $apiKey = new ApiKey($apiKeyId, $apiKeySecret);
 
-        return new Client($apiKey, $this->cacheManager, $this->cacheManagerOptions, $this->baseURL);
+        return new Client(
```

Comment:
Before creating the client, in the `build()` method, we should validate that the authentication scheme is one that we support and that is not null.